design/ceph/ceph-telemetry.md

```diff
@@ -0,0 +1,133 @@
+  }
+}
+```
+
+### Overall metrics
+- `rook/version` - Rook version.
+- `rook/kubernetes/...`
+  - `rook/kubernetes/version` - Kubernetes version.
+    - Ceph already collects os/kernel versions.
+  - `rook/kubernetes/node/count/total` - Number of Kubernetes nodes.
+- `rook/csi/...`
+  - `rook/csi/version` - Ceph CSI version.
+  - `rook/csi/plugin-pod/...` - Info about CSI plugin pods
+    - `rook/csi/plugin-pod/count/...` - Info about how many CSI plugin pods exist and of what types
+      This gives idea of how many k8s nodes are in the k8s cluster that are allowed to use Ceph
+      storage. With different types, telemetry can show if some CSI drivers are ever disabled.
+      - `rook/csi/plugin-pod/count/max` - Max number of pods of any CSI plugin type
+      - `rook/csi/plugin-pod/count/types` - Number of CSI plugin types. This can helpfully show if a
+        newer version of Rook supports a new plugin pod type than the Ceph telemetry version is
+        aware of.
+      - `rook/csi/plugin-pod/count/rbd`
+      - `rook/csi/plugin-pod/count/cephfs`
+      - `rook/csi/plugin-pod/count/nfs`
+
+
+### CephCluster metrics
+- `rook/cluster/node/count/with-ceph-pods` - Number of nodes on which the cluster has running pods.
+  - Since clusters with portable PVCs have one "node" per PVC, this will help show the actual node
+    count for Rook installs in portable environments
+  - We can get this by counting the number of crash collector pods; however, if any users disable
+    the crash collector, we will report `-1` to represent "unknown"
+- `rook/cluster/storage-class/...` - Info about storage classes related to the current Ceph cluster
+  - `rook/cluster/storage-class/count/...` - Number of storage classes of a given type
+    - `rook/cluster/storage-class/count/types` - Number of storage class types. This can helpfully
+      show if a newer version of Rook supports a new plugin pod type than the Ceph telemetry version
+      is aware of
+    - `rook/cluster/storage-class/count/total` - This is additionally useful in the case of a
+      newly-added storage class type not recognized by an older Ceph telemetry version
+    - `rook/cluster/storage-class/count/rbd`
+    - `rook/cluster/storage-class/count/cephfs`
+    - `rook/cluster/storage-class/count/nfs`
+    - `rook/cluster/storage-class/count/bucket`
+- `rook/cluster/storage/...` - Info about storage configuration
+  - `rook/cluster/storage/device-set/...` - Info about storage class device sets
+    - `rook/cluster/storage/device-set/count/...` - Number of device sets of given types
+      - `rook/cluster/storage/device-set/count/total`
+      - `rook/cluster/storage/device-set/count/portable`
+      - `rook/cluster/storage/device-set/count/non-portable`
+- `rook/cluster/mons/...` - Info about monitors and mon configuration
+  - `rook/cluster/mons/use-pvc` - true/false whether mons are on PVC
+  - `rook/cluster/mons/max-id` - The highest mon ID, which increases as mons fail over
```

Comment:
How about adding mon count?

Reply:
A bool for stretch cluster would be helpful, unless ceph already indicates that.

Reply:
Oh right, the existing ceph info should be sufficient for the mons.